src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,847 @@
+        {
+            attributes = null;
+            AddPlatformAttributes(immediateAttributes, ref attributes);
+
+            while (containingSymbol != null)
+            {
+                AddPlatformAttributes(containingSymbol.GetAttributes(), ref attributes);
+                containingSymbol = containingSymbol.ContainingSymbol;
+            }
+            return attributes != null;
+        }
+
+        private static bool AddPlatformAttributes(ImmutableArray<AttributeData> immediateAttributes, [NotNullWhen(true)] ref SmallDictionary<string, PlatformAttributes>? attributes)
+        {
+            foreach (AttributeData attribute in immediateAttributes)
+            {
+                if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name))
+                {
+                    TryAddValidAttribute(ref attributes, attribute);
+                }
+            }
+            return attributes != null;
+        }
+
+        private static bool TryFindContainingSymbolPlatformAttributes(ISymbol containingSymbol, [NotNullWhen(true)] out SmallDictionary<string, PlatformAttributes>? attributes)
+        {
+            attributes = null;
+            while (containingSymbol != null)
+            {
+                AddPlatformAttributes(containingSymbol.GetAttributes(), ref attributes);
+                containingSymbol = containingSymbol.ContainingSymbol;
+            }
+
+            return attributes != null;
+        }
+
+        private static bool TryAddValidAttribute([NotNullWhen(true)] ref SmallDictionary<string, PlatformAttributes>? attributes, AttributeData attribute)
+        {
+            if (!attribute.ConstructorArguments.IsEmpty &&
+                                attribute.ConstructorArguments[0] is { } argument &&
+                                argument.Kind == TypedConstantKind.Primitive &&
+                                argument.Type.SpecialType == SpecialType.System_String &&
+                                !argument.IsNull &&
+                                !argument.Value.Equals(string.Empty))
+            {
+                if (TryParsePlatformNameAndVersion(argument.Value.ToString(), out string platformName, out Version? version))
+                {
+                    attributes ??= new SmallDictionary<string, PlatformAttributes>(StringComparer.OrdinalIgnoreCase);
+
+                    if (!attributes.TryGetValue(platformName, out var existingAttributes))
+                    {
+                        existingAttributes = new PlatformAttributes();
+                        attributes[platformName] = existingAttributes;
+                    }
+
+                    AddAttribute(attribute.AttributeClass.Name, version, existingAttributes);
+                    return true;
+                }
+                // else report diagnostic = Diagnostic.Create(PlatformNameNullOrEmptyRule, osAttribute.ApplicationSyntaxReference.GetSyntax().GetLocation());
```

Comment:
Did you plan to add a TODO?

Reply:
or better file a tracking issue?

Reply:
At first, we were planning to warn here, but it is better to be handled with different analyzer, we have an [issue ](https://github.com/dotnet/roslyn-analyzers/issues/4014) for that, for this analyzer we gonna just ignore the invalid attribute values/hierarchy to avoid false positives, so this line will be removed